File: maxhanna.Server/Controllers/DataContracts/Files/RomMetadata.cs

namespace maxhanna.Server.Controllers.DataContracts.Files
{
  public class RomMetadata
  {
    public int? IgdbGameId { get; set; }
    public string? IgdbName { get; set; }
    public string? Summary { get; set; } 
    public long? FirstReleaseDateUnix { get; set; }
    public DateTime? FirstReleaseDate {get; set; } // This is for convenience so you don't have to convert from unix seconds on the client.

    public double? TotalRating { get; set; }
    public int? TotalRatingCount { get; set; }

    public string? CoverUrl { get; set; } 
    public string? ScreenshotsJson { get; set; }
    public string? ArtworksJson { get; set; }
    public string? VideosJson { get; set; }
    public string? PlatformsJson { get; set; }
    public string? GenresJson { get; set; }
    public int? ResetVotes { get; set; }
    public string? ActualSystem { get; set; }
  }
}

File: maxhanna.client/src/services/datacontracts/file/rom-metadata.ts

export class RomMetadata {
  igdbGameId?: number;
  igdbName?: string;
  summary?: string;

  // unix seconds
  firstReleaseDateUnix?: number;

  totalRating?: number;
  totalRatingCount?: number;

  coverUrl?: string;

  // raw json strings from server (we'll parse them in FileSearch)
  screenshotsJson?: string | string[];
  artworksJson?: string | string[];
  videosJson?: string | string[];
  // parsed convenience arrays (populated client-side)
  platformsJson?: string | string[];
  genresJson?: string | string[];
  platforms?: string[];
  genres?: string[];
  // how many users requested a reset for this enrichment
  resetVotes?: number;
  // user-selected system/core override (from rom_system_overrides table)
  actualSystem?: string;
}
